iroha/src/peer.rs

```diff
@@ -18,6 +22,247 @@ pub mod isi {
     /// synchronization components will start using it.
     #[derive(Clone, Debug, PartialEq, Io, IntoContract, Encode, Decode)]
     pub struct AddPeer {
-        pub peer: Peer,
+        pub address: String,
+        pub peer_key: PublicKey,
+    }
+}
+
+const PING_SIZE: usize = 32;
+
+#[derive(Io, Decode, Encode, Debug, Clone)]
+enum Message {
+    //TODO: introduce other features like block sync, voting and etc.
+    Ping(Ping),
+    Pong(Ping),
+    PendingTx(Transaction),
+    AddPeer(PeerId),
+    NewPeer(PeerId),
+    RemovePeer(PeerId),
+}
+
+#[derive(Encode, Decode, PartialEq, Eq, Debug, Clone, Hash)]
+struct Ping {
```

Comment:
What do you think about putting structs into enum?

Reply:
I don't think it will be useful here.